tests/integration/targets/connection/test_connection.yml

```diff
@@ -77,3 +78,15 @@
           - root
         loop_control:
           loop_var: user_name
+  
+  ### copy an empty file
+  - name: copy an empty file
+    action: "{{ action_prefix }}copy content= dest={{ remote_empty_file }}"
+  - name: stat empty file
+    action: "{{ action_prefix }}stat path={{ remote_empty_file }}"
+    register: stat_empty_file_cmd
```

Comment:
The Windows build fails at this step, but I don't fully understand, why. I believe I am using the [`win_stat` module](https://docs.ansible.com/ansible/latest/collections/ansible/windows/win_stat_module.html) correctly here. Can someone help?

```
TASK [stat empty file] *********************************************************
task path: /home/zuul/.ansible/collections/ansible_collections/community/aws/tests/integration/targets/connection/test_connection.yml:85
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible-test-11rrb__k/ansible/executor/task_executor.py", line 158, in run
    res = self._execute()
  File "/tmp/ansible-test-11rrb__k/ansible/executor/task_executor.py", line 629, in _execute
    result = self._handler.run(task_vars=vars_copy)
  File "/tmp/ansible-test-11rrb__k/ansible/plugins/action/normal.py", line 47, in run
    result = merge_hash(result, self._execute_module(task_vars=task_vars, wrap_async=wrap_async))
  File "/tmp/ansible-test-11rrb__k/ansible/plugins/action/__init__.py", line 1040, in _execute_module
    self._make_tmp_path()
  File "/tmp/ansible-test-11rrb__k/ansible/plugins/action/__init__.py", line 462, in _make_tmp_path
    result = self._low_level_execute_command(cmd, sudoable=False)
  File "/tmp/ansible-test-11rrb__k/ansible/plugins/action/__init__.py", line 1320, in _low_level_execute_command
    rc, stdout, stderr = self._connection.exec_command(cmd, in_data=in_data, sudoable=sudoable)
  File "/home/zuul/.ansible/collections/ansible_collections/community/aws/plugins/connection/aws_ssm.py", line 311, in wrapped
    return_tuple = func(self, *args, **kwargs)
  File "/home/zuul/.ansible/collections/ansible_collections/community/aws/plugins/connection/aws_ssm.py", line 539, in exec_command
    super().exec_command(cmd, in_data=in_data, sudoable=sudoable)
  File "/tmp/ansible-test-11rrb__k/ansible/plugins/connection/__init__.py", line 35, in wrapped
    self._connect()
  File "/home/zuul/.ansible/collections/ansible_collections/community/aws/plugins/connection/aws_ssm.py", line 465, in _connect
    self.start_session()
  File "/home/zuul/.ansible/collections/ansible_collections/community/aws/plugins/connection/aws_ssm.py", line 495, in start_session
    response = self._client.start_session(**start_session_args)
  File "/home/zuul/venv/lib/python3.9/site-packages/botocore/client.py", line 386, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/zuul/venv/lib/python3.9/site-packages/botocore/client.py", line 705, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.TargetNotConnected: An error occurred (TargetNotConnected) when calling the StartSession operation: i-0a4abbd44ee8abe08 is not connected.
fatal: [windows_i-0a4abbd44ee8abe08]: FAILED! => {
    "msg": "Unexpected failure during module execution: An error occurred (TargetNotConnected) when calling the StartSession operation: i-0a4abbd44ee8abe08 is not connected.",
    "stdout": ""
}
```

Reply:
The failure was `TargetNotConnected` from the `botocore` layer.  Which implies that ssm-agent disconnected on the 'host' side.

You'll see that timing-wise the Windows test was much longer (I've seen this in general with Windows).  I *suspect* (but don't have access to the logs) that the [terminator](https://github.com/mattclay/aws-terminator) started deleting the resource while the tests were still running.

Since the latest run was successful I'd not worry about that.